### PR TITLE
feat: add typescript support

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/utils/options.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ module.exports = function(options) {
     https: false,
     key: '',
     cert: '',
+    typescript: false,
```
Comment (popomore) [inline lib/utils/options.js]: 加个用例？看能否开启 egg-core

Comment (atian25) [reply lib/utils/options.js]: 加了

**Comment by atian25** [comment]
发个版本

**Comment by atian25** [comment]
1.16.0

**Pull request merged**